### PR TITLE
Implementation of Auto-Population for Address Form Fields

### PR DESCRIPTION
**Pull request by BEULAHEVANJALIN**
Card link - https://github.com/avniproject/IPH-Sickle-cell/issues/37

### CODE REVIEW

File: packages/openchs-android/src/views/form/inputComponents/LocationHierarchyInput.js [hunk 1]
```diff
@@ -17,7 +17,8 @@ class LocationHierarchyInput extends AbstractComponent {
     static propTypes = {
         concept: PropTypes.object.isRequired,
         value: PropTypes.object,
-        onSelect: PropTypes.func
+        onSelect: PropTypes.func,
+        mandatory: PropTypes.boolean
```
Comment (himeshr) [inline packages/openchs-android/src/views/form/inputComponents/LocationHierarchyInput.js:21]: Has to be PropTypes.bool

File: packages/openchs-android/src/views/form/inputComponents/LocationHierarchyInput.js [hunk 2]
```diff
@@ -46,6 +47,7 @@ class LocationHierarchyInput extends AbstractComponent {
             minLevelTypeUUIDs={this.minLevelTypeUUIDs()}
             maxLevelTypeUUID={this.maxLevelTypeUUID()}
             isOutsideCatchment={!this.isWithinCatchment()}
+            mandatory={this.props.mandatory}
```
Comment (himeshr) [inline packages/openchs-android/src/views/form/inputComponents/LocationHierarchyInput.js:50]: Need to handle for AutocompleteSearch

**Comment by himeshr** [comment]
As discussed with @BEULAHEVANJALIN we'll have to spend some more time validating feature completeness.

**Comment by mahalakshme** [comment]
@himeshr Also the validity of having auto selection on Location concept type also needs to be verified. I will analyse it.

**Comment by himeshr** [comment]
> @himeshr Also the validity of having auto selection on Location concept type also needs to be verified. I will analyse it.

Auto-selection is to happen, only if its a Mandatory Location concept FormElement, with a single value at an AddressLevelType.

**Pull request closed without merging**